node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/NodePrioritizer.java

```diff
@@ -77,11 +83,13 @@
      *
      * @return hostname or null if not able to do the lookup
      */
-    private static String lookupHostname(String ipAddress) {
+    static String lookupHostname(String ipAddress) {
```

Comment:
Why does this class have it's own lookup method? Consider using `DnsNameResolver` and extending it with the method you need. `DnsNameResolver` always ask external DNS, so incorrect entries in `/etc/hosts` won't affect the result.

There's also a `MockNameResolver` so you don't need to do the `if (unitTesting)` stuff.